### PR TITLE
Add users

### PR DESCRIPTION
**Pull request by mainlyetcetera**
- add sad path testing for `sessions_spec.rb`
  - testing whether non-input user results in error
- merges changes from add-key, main 
  - checked by `.travis.yml` existing on this branch

### CODE REVIEW

**Comment by mainlyetcetera** [comment]
Have any thoughts on why this build would be failing? I believe I have merged the `.travis.yml` successfully.

**Comment by ribbansmax** [comment]
> Have any thoughts on why this build would be failing? I believe I have merged the `.travis.yml` successfully.

I think it has something to do with the gemfile
I'll try pulling down, deleting the lock, bundle installing and repushing

**Comment by ribbansmax** [comment]
That was it

File: spec/requests/api/v1/sessions_spec.rb [hunk 1]
```diff
@@ -20,11 +20,33 @@
 
       data = JSON.parse(response.body, symbolize_names: true)[:data]
 
+      expect(User.last.api_key).to eq(data[:attributes][:api_key])
       expect(data[:attributes][:username]).to eq(user.username)
     end
   end
 
   describe 'sad' do
+    it 'should not find user not there' do
```
Comment (ribbansmax) [inline spec/requests/api/v1/sessions_spec.rb:29]: This is good, although we will need more sad path tests like:

- Incorrect password
- mismatch on password and confirmation
- Empty username

File: app/controllers/api/v1/sessions_controller.rb [hunk 2]
```diff
@@ -6,9 +6,9 @@ def create
       render json: UserSerializer.new(@user)
     else
       render json: { 
-        status: 401,
+        # do we need logged_in: false here?
```
Comment (ribbansmax) [inline app/controllers/api/v1/sessions_controller.rb:9]: I don't think so, as far As I could work out, that would be the default on the Front End...

File: spec/requests/api/v1/sessions_spec.rb (hunk 1 above)
Comment (mainlyetcetera) [reply spec/requests/api/v1/sessions_spec.rb:29]: Agreed. The push so far was mostly a proof-of-concept, making sure that the build could run after the merges.

File: app/controllers/api/v1/sessions_controller.rb [hunk 3]
```diff
@@ -21,6 +21,7 @@ def is_logged_in?
     else
       render json: {
         logged_in: false,
+        # should this msg say 'not logged in' instead?
```
Comment (ribbansmax) [inline app/controllers/api/v1/sessions_controller.rb:24]: No- this is for the FE as an error message, so this is more helpful I think

File: app/controllers/api/v1/sessions_controller.rb (hunk 2 above)
Comment (mainlyetcetera) [reply app/controllers/api/v1/sessions_controller.rb:9]: Ok. This is mostly my unfamiliarity with `Ruby`, and where all we need to account for this.

**Review by ribbansmax — COMMENTED** [review]
We will also need tests for the other sessions methods, the logout and the is_logged_in?

File: spec/requests/api/v1/sessions_spec.rb (hunk 1 above)
Comment (mainlyetcetera) [reply spec/requests/api/v1/sessions_spec.rb:29]: The mismatch on password/confirmation are more when the `user` is created, and incorrect password is during actual login, right? Want to make sure I'm matching where you see these are all applied.

**Pull request merged**